src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -138,6 +138,36 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                                         printPackages = false;
                                     }
                                 }
+                                // Handle --deprecated
+                                else if (listPackageArgs.IncludeDeprecated)
+                                {
+                                    await GetDeprecationInfoAsync(packages, listPackageArgs);
+
+                                    var noPackagesLeft = true;
+
+                                    // Filter the packages for deprecated
+                                    foreach (var frameworkPackages in packages)
+                                    {
+                                        frameworkPackages.TopLevelPackages = frameworkPackages.TopLevelPackages
+                                            .Where(p => p.ResolvedPackageMetadata.DeprecationMetadata != null);
+
+                                        frameworkPackages.TransitivePackages = frameworkPackages.TransitivePackages
+                                            .Where(p => p.ResolvedPackageMetadata.DeprecationMetadata != null);
+
+                                        if (frameworkPackages.TopLevelPackages.Any() || frameworkPackages.TransitivePackages.Any())
```

Comment:
I think we only look at top level packages to preserve the behavior as in outdated.

Reply:
Or at least review it with Anand :)

Reply:
Fixed in https://github.com/NuGet/NuGet.Client/pull/2959/commits/2a1eacc2cd3c3b8568b5430afdc73542306cc9be